File: frontend/src/types/benefits.ts
export interface BenefitProgram {
  id: string;
  name: string;
  type: "federal" | "state" | "local" | "nonprofit";
  category: "insurance" | "prescription" | "clinic" | "financial_assistance";
  description: string;
  eligibility_summary: string;
  income_limit_fpl_percent: number | null;
  age_min: number | null;
  age_max: number | null;
  requires_children: boolean;
  requires_disability: boolean;
  requires_pregnancy: boolean;
  states: string[];
  application_url: string;
  phone: string;
  estimated_savings: string;
}

export interface EligibilityInput {
  annual_income: number;
  household_size: number;
  state: string;
  age: number;
  has_children: boolean;
  is_pregnant: boolean;
  has_disability: boolean;
  currently_insured: boolean;
  zip_code: string;
}

export interface EligibilityResult {
  eligible_programs: BenefitProgram[];
  nearby_health_centers: HealthCenter[];
  fpl_percentage: number;
  summary: string;
}

export interface HealthCenter {
  name: string;
  address: string;
  phone: string;
  distance_miles: number;
  services: string[];
  sliding_scale: boolean;
  website: string;
}

File: backend/app/schemas/benefits.py
from pydantic import BaseModel
from typing import Optional


class BenefitProgram(BaseModel):
    id: str
    name: str
    type: str
    category: str
    description: str
    eligibility_summary: str
    income_limit_fpl_percent: Optional[float] = None
    age_min: Optional[int] = None
    age_max: Optional[int] = None
    requires_children: bool
    requires_disability: bool
    requires_pregnancy: bool
    states: list[str]
    application_url: str
    phone: str
    estimated_savings: str


class HealthCenter(BaseModel):
    name: str
    address: str
    phone: str
    distance_miles: float
    services: list[str]
    sliding_scale: bool
    website: str


class EligibilityInput(BaseModel):
    annual_income: float
    household_size: int
    state: str
    age: int
    has_children: bool = False
    is_pregnant: bool = False
    has_disability: bool = False
    currently_insured: bool = False
    zip_code: str = ""


class EligibilityResult(BaseModel):
    eligible_programs: list[BenefitProgram]
    nearby_health_centers: list[HealthCenter]
    fpl_percentage: float
    summary: str
